src/Listener/ExceptionListener.php

```diff
@@ -154,9 +154,9 @@ public function getErrorCodePage($statusCode)
      *
      * @throws \Exception
      */
-    public function onKernelException(GetResponseForExceptionEvent $event): void
```

Comment:
@Hanmac you should provide some support to BC or should he just add a comment in UPGRADE.md @VincentLanglet ?

Note: the same for the others methods that you changed the attribute's type

Reply:
i made that MR extra for 4.x because it is a BC break
and it is done to make 4.x 

or do you mean i should make another PR for 3.x?
The ExceptionListener should be internal stuff imo?

Reply:
I don't know, if we should provide BC support for 3.x or just add a comment in Upgrade.md

as this class is not final, if someone extend this class and override this method, it'll break in `4.x`, I guess you need to provide some support from 3.x or add deprecation for whom extends, have noticed that it'll be changed, for example

```php
/**
 * NEXT_MAJOR: Restrict the $event param with  "ExceptionEvent" type
 *
 * @param GetResponseForExceptionEvent|ExceptionEvent $event
 */
public function onKernelException($event): void
{
    //NEXT_MAJOR: Remove this if
    if ($event instanceof GetResponseForExceptionEvent) {
        //trigger a deprecation message
    }
}
```

and you open this for 3.x branch and remove the old code in `4.x`

Reply:
well if I understand correct, we should provide something to support a smooth upgrade from `3.x` to `4.x`
https://symfony.com/doc/current/contributing/code/bc.html

then you need to do something like the example that I added above!

Reply:
i also could do a small PR for 3.x too, but that is for @VincentLanglet do decide what to do

Reply:
> I don't know, if we should provide BC support for 3.x or just add a comment in Upgrade.md
> 
> as this class is not final, if someone extend this class and override this method, it'll break in `4.x`, I guess you need to provide some support from 3.x and add deprecation messages for whom extends, have noticed that it'll be changed, for example
> 
> ```
> /**
>  * NEXT_MAJOR: Restrict the $event param with  "ExceptionEvent" type
>  *
>  * @param GetResponseForExceptionEvent|ExceptionEvent $event
>  */
> public function onKernelException($event): void
> {
>     //NEXT_MAJOR: Remove this "if"
>     if ($event instanceof GetResponseForExceptionEvent) {
>         //trigger a deprecation message
>     }
> }
> ```
> 
> and you open this for 3.x branch and remove the old code in `4.x`

it won't work too, it'll leads warnings too 😄

Reply:
If I understand correctly, if we drop 7.3 support this is BC. isn't it ?
Let's drop 7.3 then.

Can you bump the php version to 7.4 ?